Change test_torchinductor_opinfo.py to mark skips/xfails in a better way

fdrocha opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94813

With this change, expected failures will be correctly reported as such by pytest (instead of passes as before).
It was sometimes a little confusing to see operators you did not expect to work in inductor reported as passing their tests.

One downside is that expected failures/skips for test variants have now to be identified by tuples. I.e., `("max", "reduction_no_dim"): {f16},` instead of just `"max.reduction_no_dim": {f16}`. It seems to me it is worth it.

This change would also allow to simplify `TestInductorOpInfo` class a little, since it doesn't have to handle the skips/xfails anymore, but that might require dropping support for things like `PYTORCH_COLLECT_EXPECT` and `PYTORCH_FAIL_ON_SUCCESS` so I didn't do it.

Also couple of other minor changes:

 - Got rid of c32, c64, c128 in torchinductor_opinfo. We don't support complex numbers, so they shouldn't be necessary.
 - Renamed TestExpect Enum to ExpectedTestResult to get rid of a pytest warning that thinks it is a class that has tests.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

Commit:
Get rid of c32, c64, c128 in torchinductor_opinfo

We don't support complex numbers, so they shouldn't be necessary.
For same reason, moved some opts that return complex values to
skips from xfails.

[ghstack-poisoned]

Commit:
Update on "Get rid of c32, c64, c128 in torchinductor_opinfo"


We don't support complex numbers, so they shouldn't be necessary.
For same reason, moved some opts that return complex values to
skips from xfails.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

voznesenskym commented:
There is value in not skipping them, if only to catch a small window of bugs where we do not support an op, but it does not segfault. A good number of xfails have no plan for enablement, and a  good number of skips are for things that cannot run at all, fail or otherwise. I'm not sure moving these ops to a "worse" list is a good change.

fdrocha commented:
> There is value in not skipping them, if only to catch a small window of bugs where we do not support an op, but it does not segfault. A good number of xfails have no plan for enablement, and a good number of skips are for things that cannot run at all, fail or otherwise. I'm not sure moving these ops to a "worse" list is a good change.

That's all reasonable, I'll move them back.

I think what made me want to move them is that pytest is reporting all the expected failures as PASS, instead of XFAIL, and that can be confusing. At least with skip it's clear they are not passing. But the real fix is to have pytest report the status accurately, I'll see if there's a nice way of doing that.

Commit:
Update on "A couple of small clean ups to test_torchinductor_opinfo.py"


Namely, get rid of c32, c64, c128 in torchinductor_opinfo. We don't support complex numbers, so they shouldn't be necessary.

For same reason, moved some opts that return complex values to
skips from xfails.

Also, renamed TestExpect Enum to ExpectedTestResult to get rid of a pytest warning that think it is a class that has tests.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

lezcano commented on test/inductor/test_torchinductor_opinfo.py:
```diff
@@ -267,7 +223,33 @@ def process(device_type):
     "view_as_complex": {f16},
-    "norm.inf": {f16},
+    ("norm", "inf"): {f16},
+    # fft ops sometimes succeed locally and fail on CI.
+    # they return complex values which is known unsupported,
+    # so there is not much point in testing them currently.
```
Is this comment now stale?

fdrocha replied:
I think you are right, removed it.

lezcano commented on test/inductor/test_torchinductor_opinfo.py:
```diff
@@ -54,16 +55,13 @@
 
 # Success forces pass; failure forces fail; skip unconditionally skips testing
-TestExpect = Enum("TestExpect", ("SUCCESS", "XFAILURE", "SKIP"))
+ExpectedTestResult = Enum("ExpectedTestResult", ("SUCCESS", "XFAILURE", "SKIP"))
```
Why this change?

lezcano replied:
nvm, just saw the OP

lezcano reviewed (approved):
Changing the name to a tuple LGTM. We already do that in tests within functorch.

Commit:
Update on "Change test_torchinductor_opinfo.py to mark skips/xfails in a better way"


With this change, expected failures will be correctly reported as such by pytest (instead of passes as before).
It was sometimes a little confusing to see operators you did not expect to work in inductor reported as passing their tests.

One downside is that expected failures/skips for test variants have now to be identified by tuples. I.e., `("max", "reduction_no_dim"): {f16},` instead of just `"max.reduction_no_dim": {f16}`. It seems to me it is worth it.

This change would also allow to simplify `TestInductorOpInfo` class a little, since it doesn't have to handle the skips/xfails anymore, but that might require dropping support for things like `PYTORCH_COLLECT_EXPECT` and `PYTORCH_FAIL_ON_SUCCESS` so I didn't do it.

Also couple of other minor changes:

 - Got rid of c32, c64, c128 in torchinductor_opinfo. We don't support complex numbers, so they shouldn't be necessary.
 - Renamed TestExpect Enum to ExpectedTestResult to get rid of a pytest warning that think it is a class that has tests.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Commit:
Update on "Change test_torchinductor_opinfo.py to mark skips/xfails in a better way"


With this change, expected failures will be correctly reported as such by pytest (instead of passes as before).
It was sometimes a little confusing to see operators you did not expect to work in inductor reported as passing their tests.

One downside is that expected failures/skips for test variants have now to be identified by tuples. I.e., `("max", "reduction_no_dim"): {f16},` instead of just `"max.reduction_no_dim": {f16}`. It seems to me it is worth it.

This change would also allow to simplify `TestInductorOpInfo` class a little, since it doesn't have to handle the skips/xfails anymore, but that might require dropping support for things like `PYTORCH_COLLECT_EXPECT` and `PYTORCH_FAIL_ON_SUCCESS` so I didn't do it.

Also couple of other minor changes:

 - Got rid of c32, c64, c128 in torchinductor_opinfo. We don't support complex numbers, so they shouldn't be necessary.
 - Renamed TestExpect Enum to ExpectedTestResult to get rid of a pytest warning that think it is a class that has tests.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

fdrocha commented:
@pytorchbot merge

huydhn commented:
@pytorchbot revert -m 'Sorry for reverting your PR, but it causes failures on trunk https://hud.pytorch.org/pytorch/pytorch/commit/bfc0d5e22c34e5888c394735bf696e2f45e07816 due to a landrace with https://github.com/pytorch/pytorch/commit/b6df9876719f223670103e7c6049687ac75c3def' -c landrace

huydhn commented:
@pytorchbot rebase

Commit:
Update on "Change test_torchinductor_opinfo.py to mark skips/xfails in a better way"


With this change, expected failures will be correctly reported as such by pytest (instead of passes as before).
It was sometimes a little confusing to see operators you did not expect to work in inductor reported as passing their tests.

One downside is that expected failures/skips for test variants have now to be identified by tuples. I.e., `("max", "reduction_no_dim"): {f16},` instead of just `"max.reduction_no_dim": {f16}`. It seems to me it is worth it.

This change would also allow to simplify `TestInductorOpInfo` class a little, since it doesn't have to handle the skips/xfails anymore, but that might require dropping support for things like `PYTORCH_COLLECT_EXPECT` and `PYTORCH_FAIL_ON_SUCCESS` so I didn't do it.

Also couple of other minor changes:

 - Got rid of c32, c64, c128 in torchinductor_opinfo. We don't support complex numbers, so they shouldn't be necessary.
 - Renamed TestExpect Enum to ExpectedTestResult to get rid of a pytest warning that thinks it is a class that has tests.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Commit:
Update on "Change test_torchinductor_opinfo.py to mark skips/xfails in a better way"


With this change, expected failures will be correctly reported as such by pytest (instead of passes as before).
It was sometimes a little confusing to see operators you did not expect to work in inductor reported as passing their tests.

One downside is that expected failures/skips for test variants have now to be identified by tuples. I.e., `("max", "reduction_no_dim"): {f16},` instead of just `"max.reduction_no_dim": {f16}`. It seems to me it is worth it.

This change would also allow to simplify `TestInductorOpInfo` class a little, since it doesn't have to handle the skips/xfails anymore, but that might require dropping support for things like `PYTORCH_COLLECT_EXPECT` and `PYTORCH_FAIL_ON_SUCCESS` so I didn't do it.

Also couple of other minor changes:

 - Got rid of c32, c64, c128 in torchinductor_opinfo. We don't support complex numbers, so they shouldn't be necessary.
 - Renamed TestExpect Enum to ExpectedTestResult to get rid of a pytest warning that thinks it is a class that has tests.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

fdrocha commented:
@pytorchbot merge
